base/setup/lib/fsutil.c

```diff
@@ -688,6 +707,89 @@ InstallBtrfsBootCode(
     return Status;
 }
 
+NTSTATUS
+InstallExt2BootCode(
+    IN PCWSTR SrcPath,          // Ext2 bootsector source file (on the installation medium)
+    IN HANDLE DstPath,          // Where to save the bootsector built from the source + partition information
+    IN HANDLE RootPartition)    // Partition holding the (old) FAT32 information
+{
+    NTSTATUS Status;
+    UNICODE_STRING Name;
+    IO_STATUS_BLOCK IoStatusBlock;
+    LARGE_INTEGER FileOffset;
+    BOOTCODE OrigBootSector = {0};
+    BOOTCODE NewBootSector  = {0};
+
+    /* Allocate and read the current original partition bootsector */
+    Status = ReadBootCodeByHandle(&OrigBootSector,
+                                  RootPartition,
+                                  EXT2_BOOTSECTOR_SIZE);
+    if (!NT_SUCCESS(Status))
+        return Status;
+
+    /* Allocate and read the new bootsector (2 sectors) from SrcPath */
+    RtlInitUnicodeString(&Name, SrcPath);
+    Status = ReadBootCodeFromFile(&NewBootSector,
+                                  &Name,
+                                  EXT2_BOOTSECTOR_SIZE);
+    if (!NT_SUCCESS(Status))
+    {
+        FreeBootCode(&OrigBootSector);
+        return Status;
+    }
+
+    /* Adjust bootsector (copy old BPB) */
+    ((PEXT2_BOOTSECTOR)NewBootSector.BootCode)->BootDrive = 0xff;
+    ((PEXT2_BOOTSECTOR)NewBootSector.BootCode)->BootPartition = 0x00;
+    ((PEXT2_BOOTSECTOR)NewBootSector.BootCode)->Ext2VolumeStartSector = ((PEXT2_BOOTSECTOR)OrigBootSector.BootCode)->Ext2VolumeStartSector;
```

Comment:
This information does not exist you will need to get the start sector from MBR or something